src/main/java/com/adobe/target/edge/client/ondevice/OnDeviceDecisioningService.java

```diff
@@ -200,12 +196,13 @@ public TargetDeliveryResponse executeRequest(TargetDeliveryRequest deliveryReque
         targetResponse.getResponse().getExecute(),
         notifications);
 
-    TelemetryEntry telemetryEntry =
-        createTelemetryEntry(
-            deliveryRequest, targetResponse, timer.timeEnd(TIMING_EXECUTE_REQUEST));
-    telemetry.addEntriesItem(telemetryEntry);
+    Telemetry telemetry = new Telemetry();
+    TelemetryEntry telemetryEntry = telemetryService.createTelemetryEntry(deliveryRequest, targetResponse, timer.timeEnd(TIMING_EXECUTE_REQUEST));
+    if(telemetryEntry!=null) {
```

Comment:
Surprised our Spotless plugin didn't catch this, but please make this `if(telemetryEntry != null)` to be consistent with other places in the code.